src/components/application_manager/src/rpc_passing_handler.cc

```diff
@@ -254,34 +254,73 @@ bool RPCPassingHandler::CycleThroughRPCPassingRequests(
 void RPCPassingHandler::OnPassThroughRequestTimeout() {
   LOG4CXX_DEBUG(logger_, "RPC Passing request timeout");
   auto timeout_entry = timeout_queue_.front();
-  uint32_t correlation_id = std::get<1>(timeout_entry);
-  timeout_queue_.pop();
+  uint32_t correlation_id = timeout_entry.second;
+  RemoveTimeoutTimer(correlation_id);
   LOG4CXX_DEBUG(logger_,
                 "Cycling through requests for correlation_id "
                     << correlation_id);
   CycleThroughRPCPassingRequests(correlation_id);
+  if (request_queue_[correlation_id].empty()) {
```

Comment:
I can't help but notice that this code is always following `CycleThroughRPCPassingRequests`, it should probably just be in that function

Reply:
Fixed 👍